ecosystem/sep-0012.md

```diff
@@ -475,9 +475,107 @@ Anchor has no information on the customer | `404 Not Found`
 
 Allow the wallet to provide a callback URL to the anchor. The provided callback URL will replace (and supercede) any previously-set callback URL for this account.
 
-Whenever the user's `status` field changes, the anchor will issue a POST request to the callback URL. The payload of the POST request will be the same as the response of [`GET /customer`](#customer-get).
+Whenever the user's `status` field changes, the anchor will issue a POST request to the callback URL. 
+The payload of the POST request will be the same as the response of [`GET /customer`](#customer-get) with an additional field `timestamp` set to the current timestamp at the time the Anchor issue the POST request. This is used to assure the freshness of the request and to prevent this request to be replayed in the future. Additionally the Anchor will provide a base64 signature of the POST request payload in the HTTP Header `X-Stellar-Signature`.
```

Comment:
Yes that's why I added a timestamp but the relay is still possible if executed in a short timeframe.
I think I understand what you mean by signing the host header now. I thought about it the other way, in my mind the host was set to the KYC SEP-12 Server whereas what you suggest is to have a host value set to the destination wallet correct?
Something like `X-Stellar-Signature: t=<timestamp>, h=api.vibrantapp.com, s=<base64 signature>` ?
The SEP says that the callback needs to use the parameters of the [`Customer GET`](https://github.com/C0x41lch0x41/stellar-protocol/blob/patch-2/ecosystem/sep-0012.md#customer-get) endpoint. One of parameter is `account` (and optionally `memo`). It is optional since the endpoint uses SEP-10 JWT token but we could say this is mandatory here.
So that:
- `timestamp` prevents replay
- `host` prevents relay from one wallet backend to another
- `account` prevents ID clashing

Reply:
If I'm understanding correctly the callback the HTTP host will be the wallet server, since the SEP-12 server will be sending a request to the wallet server.

Reply:
Yes exactly. I was think on including it in the X-Stellar-Signature header as well but maybe that's not necessary.